modules/account-lib/src/coin/sol/transaction.ts

```diff
@@ -188,4 +189,75 @@ export class Transaction extends BaseTransaction {
     this._outputs = outputs;
     this._inputs = inputs;
   }
+
+  /** @inheritDoc */
+  explainTransaction(): TransactionExplanation {
+    const decodedInstructions = instructionParamsFactory(this._type, this._solTransaction.instructions);
+
+    let memo: string | undefined = undefined;
+    let durableNonce: DurableNonceParams | undefined = undefined;
+
+    let outputAmount = new BigNumber(0);
+    const outputs: TransactionRecipient[] = [];
+
+    for (const instruction of decodedInstructions) {
+      switch (instruction.type) {
+        case InstructionBuilderTypes.NonceAdvance:
+          durableNonce = (instruction as Nonce).params;
+          break;
+        case InstructionBuilderTypes.Memo:
+          memo = (instruction as Memo).params.memo;
+          break;
+        case InstructionBuilderTypes.Transfer:
+          const transferInstruction = instruction as Transfer;
+          outputs.push({
+            address: transferInstruction.params.toAddress,
+            amount: transferInstruction.params.amount,
+          });
+          outputAmount = outputAmount.plus(transferInstruction.params.amount);
+          break;
+        case InstructionBuilderTypes.CreateNonceAccount:
+          const createInstruction = instruction as WalletInit;
+          outputs.push({
+            address: createInstruction.params.nonceAddress,
+            amount: createInstruction.params.amount,
+          });
+          outputAmount = outputAmount.plus(createInstruction.params.amount);
+          break;
+        default:
+          continue;
+      }
+    }
+
+    const explainedTransaction = {
+      displayOrder: [
+        'id',
+        'type',
+        'blockhash',
+        'durableNonce',
+        'outputAmount',
+        'changeAmount',
+        'outputs',
+        'changeOutputs',
+        'fee',
+        'memo',
+      ],
+      // TODO (STLX-8791): Need to figure out what id to use for unsigned transactions.
+      id: this.id || 'undefined',
+      type: TransactionType[this.type].toString(),
+      changeOutputs: [],
+      changeAmount: '0',
+      outputAmount: outputAmount.toFixed(0),
+      outputs: outputs,
+      fee: {
+        // TODO(STLX-8791): need to figure out how to get fee information.
+        fee: '0',
```

Comment:
With just a rawTx there is no way to figure out the fee, we only can extract the number of signatures that is needed to calculate the fee.

Reply:
that's true... will ask around to see how we want to handle this for the deserialization flow. For Dot, i see that they have clients set the `fee`, but not sure how they're getting it back on deserialization.